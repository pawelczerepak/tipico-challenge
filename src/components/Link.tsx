import styled from 'styled-components';

export default styled.a`
  text-decoration: none;
  color: #c8102e;
  &:visited {
    color: #91102c;
  }
`;
