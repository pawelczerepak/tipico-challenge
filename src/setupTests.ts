import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';
import client from './apiClient';

configure({ adapter: new Adapter() });

export const mockStore = configureMockStore([thunk, axiosMiddleware(client)]);
