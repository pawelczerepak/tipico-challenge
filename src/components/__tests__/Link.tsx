import React from 'react';
import '../../setupTests';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Link from '../Link';

describe('Link', () => {
    it('capture snapshot', () => {
        expect(toJson(shallow(
            <Link/>,
        )))
            .toMatchSnapshot();
    });
});
