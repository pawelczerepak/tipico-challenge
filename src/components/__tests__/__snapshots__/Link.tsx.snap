// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link capture snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  text-decoration: none;
  color: #c8102e;
  &:visited {
    color: #91102c;
  }
",
        ],
      },
      "displayName": "styled.a",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "a",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
